plotly/plotly/plotly.py

```diff
@@ -105,6 +108,21 @@ def _plot_option_logic(plot_options):
             'fileopt' not in session_plot_options and
             'fileopt' not in plot_options):
         current_plot_options['fileopt'] = 'overwrite'
+
+    if ('world_readable' not in plot_options and
+            plot_options['share_key_enabled']):
+        current_plot_options['world_readable'] = False
+        current_plot_options['share_key_enabled'] = True
+
+    # If plot set as public but Share_key_enabled is activated 
+    if ('world_readable' in plot_options and plot_options['world_readable']
+            and plot_options['share_key_enabled']):
+        current_plot_options['world_readable'] = True
+        current_plot_options['share_key_enabled'] = False
+        warnings.warn(
+            "Looks like you've activated the share_key for a public plot. "
+            "share_key can only be activaited for private plots."
+            "\nQuestions? support@plot.ly")
```

Comment:
:cow2: This isn't your fault, but should really be changed. `support@plot.ly` is intended for enterprise-related emails, not for the general public consumers. We should really be pointing users to `feedback@plot.ly`. We do this in a number of places.

@chriddyp actually, we could be a bit sneaky and return different *supporty* emails based on whether the plotly_domain matches the default plotly_domain in the session config? too sneaky? worthwhile?